### PR TITLE
Improve `writeCheckpoint` in MySQL storage implementation

### PR DESCRIPTION
**Pull request by roger2hk**
#21

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -56,7 +56,7 @@ require (
 	go.opentelemetry.io/otel v1.24.0 // indirect
 	go.opentelemetry.io/otel/metric v1.24.0 // indirect
 	go.opentelemetry.io/otel/trace v1.24.0 // indirect
-	golang.org/x/crypto v0.25.0 // indirect
+	golang.org/x/crypto v0.25.0
```
Comment (mhutchinson) [inline go.mod:59]: I don't see how this change could have caused this. Did you run `go mod tidy`?

File: storage/mysql/mysql.go [hunk 2]
```diff
@@ -55,10 +60,25 @@ func New(ctx context.Context, db *sql.DB) (*Storage, error) {
...
+		if err != nil {
+			return nil, err
+		}
+		// Defer a rollback in case anything fails.
+		defer func() {
+			if err := tx.Rollback(); err != nil && err != sql.ErrTxDone {
+				klog.Errorf("Failed to rollback in write initial checkpoint: %v", err)
+			}
+		}()
+		if err := s.writeCheckpoint(ctx, tx, 0, []byte("")); err != nil {
```
Comment (mhutchinson) [inline storage/mysql/mysql.go:74]: Is `[]byte("")` the right value? That's the empty byte string, where it should be a _hash_ of the empty bytes. See https://github.com/transparency-dev/trillian-tessera/blob/main/cmd/example-posix/main.go#L102C24-L102C57 where it uses `rfc6962.DefaultHasher.EmptyRoot()`

File: go.mod (hunk 1 above)
Comment (roger2hk) [reply go.mod:59]: Yes, I did.

File: storage/mysql/mysql.go (hunk 2 above)
Comment (roger2hk) [reply storage/mysql/mysql.go]: Updated to `rfc6962.DefaultHasher.EmptyRoot()`, thanks!

**Review by mhutchinson — APPROVED** [review]

**Pull request merged**